torch/csrc/distributed/c10d/reducer.hpp

```diff
@@ -175,27 +181,34 @@ class TORCH_API Reducer {
 
   // Rebuild buckets based on rebuilt_params_ and rebuilt_param_indices_
   // according to when tensors received grads in the backward pass.
-  // TODO this function makes broadcast communication call and
-  // could be overlapped with next forward() call, thus
-  // it could be async. Will make it async when rebuilding buckets for
-  // find_unused_parameters = true case, as we could rebuild buckets more than
-  // once for find_unused_parameters = true case, where subgraphs are trained
-  // and parameter indices order may change more frequently.
-  // For find_unused_parameters = false case, buckets are only rebuilt once,
-  // the performance cost is negligible. Returns true if the buckets were
-  // rebuilt.
+  // Returns true if the buckets were rebuilt.
   bool rebuild_buckets();
 
   // Install futures that should be awaited at end of backwards. Currently these
-  // are only used by user-defined custom buffer reduction hooks, but can be generalized
-  // to any user-originating futures that need to be awaited.
+  // are only used by user-defined custom buffer reduction hooks, but can be
+  // generalized to any user-originating futures that need to be awaited.
   void install_futures(c10::List<c10::intrusive_ptr<c10::ivalue::Future>> futs);
 
   // Returns true if we should rebuild buckets, else false. We only rebuild
-  // buckets once after the first iteration and never rebuild them if
-  // find_unused_parameters_.
+  // buckets once after the first iteration.
+  // We always rebuild bucket when find_unused_parameters=False, as graph is
+  // static when find_unused_parameters=False.
```

Comment:
Idea: should we refactor Reducer implementation to just call into static_graph code if find_unused_parameters=False? 

If they are equivalent now, this further implies that static_graph=True can be a default setting in DDP, without any BC breakages, because currently DDP is defaulted to find_unused=False. And when find_unused_parameters=True we could enforce static_graph=False. Should we try to flip this flag?